gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/factories/langchain_factory.py

```diff
@@ -188,7 +242,40 @@ def create_observability_callback_handler(
         The Observability Callback Handler
     """
     if observability_setting is not None:
-        return get_callback_handler_factory(setting=observability_setting).get_callback_handler(
-            trace_name=trace_name.value)
+        return get_callback_handler_factory(
+            setting=observability_setting
+        ).get_callback_handler(trace_name=trace_name.value)
 
     return None
+
+
+def get_guardrail_factory(setting: BaseGuardrailSetting) -> GuardrailFactory:
+    """
+    Retrieves the appropriate GuardrailFactory instance based on the provided setting.
+
+    Parameters
+    ----------
+    setting : BaseGuardrailSetting
+        An instance of a setting class that specifies the configuration for the GuardrailFactory.
+
+    Returns
+    -------
+    GuardrailFactory
+        An instance of a GuardrailFactory that matches the given setting.
+    """
```

Comment:
```suggestion
    """
    Retrieves the appropriate GuardrailFactory instance based on the provided setting.
        Args:
        setting : The Guardrail Setting
    Returns:
        The Guardrail Factory, or raise an exception otherwise
    """
```